portfolio/src/main/webapp/script.js

```diff
@@ -292,3 +292,30 @@ function showMarkerText(marker, descriptions) {
     $("#map-text").text(descriptions[marker.title]);
   } 
 }
+
+/**
+ * Function gets called when the language selected changes. 
+ * Translated all id's in a list to the selected language
+ */
+async function onLangSelect() {
+  const idsToTranslate = ["bio-title", "bio", "bio-supplement"]
+  const languageCode = $("#lang-select").val();
+  idsToTranslate.forEach(async id => {
+    const text = $(`#${id}`).text();
+    const params = new URLSearchParams();
+    params.append('languageCode', languageCode);
+    params.append('text', text);
+    const resp = await fetch('/translate', {method: 'POST', body: params});
+    const respText = await resp.text();
+    replaceText(id, respText);
+  })
+}
+
+/**
+ * Replaces test at an id with some new text. 
+ * id refers to the element id and newTest refers to the text to replace the old one with
+ */
+function replaceText(id, newText) {
+  $(`#${id}`).text("");
```

Comment:
Do you have to clear the text first? If not, you can also consider inlining this function (aka not have a function call but directly write the body in the caller).

Reply:
oh! I don't haha just just inlined it